lib/common/xml_attr.c

```diff
@@ -78,13 +78,15 @@ pcmk__dump_xml_attr(const xmlAttr *attr, GString *buffer)
 
     name = (const char *) attr->name;
     value = (const char *) attr->children->content;
+    if (value == NULL) {
+        return;
+    }
```

Comment:
An empty value (`<e a=""/>`) would be better in case the XML has a schema that requires the attribute

Reply:
That's a reasonable concern, but an empty string is a value isn't it? The flip side of "there may be a schema that requires this" is:

"There may be a caller that treats an absent attribute differently from an attribute whose value is the empty string, and the relevant schema doesn't require the attribute."

Reply:
Hmm, less than ideal either way. I suppose it doesn't matter which we pick. Maybe add a comment describing the choice.